### ISSUE TITLE
Powerlevel10k is suddenly broken in Intellij IDEA terminal

### ISSUE THREAD

**Issue by NovemberZulu** [romkatv/powerlevel10k#2682]
It seems to happen after I updated macOS to 14.5, but I am not sure. At least I don't recall updates in IDEA, I am running 2024.1.1U. I tried tweaking IDEA settings, but no luck. It works fine in iterm2, but totally broken in IDEA, see screenshots. Any ideas where to look? Thank you!
![iterm2](https://github.com/romkatv/powerlevel10k/assets/897004/e991d8f2-d482-4680-a29d-146f68614324)
![IDEA](https://github.com/romkatv/powerlevel10k/assets/897004/64224952-7da2-4ede-9e7d-dbc79a464b9d)

**Comment by liond92** [comment]
I face the same issues :( 
I am using macOs 14.1.1  and IDEA 2024.1.1

**Comment by daniel-gut** [comment]
As a temporary workaround to have a clean terminal in intellij, I change the theme for IDEA by updating `.zshrc`.

```
if [[ $IDEA_INITIAL_DIRECTORY ]]; then
  ZSH_THEME="robbyrussell"
else
  ZSH_THEME="powerlevel10k/powerlevel10k"
fi
```

Every other ENV var which is present in intellij works as well. Also other themes might work just fine.

**Comment by SuCicada** [comment]
same.  In my case, I installed the fig plug-in before, which caused it to break. 
So I uninstalled fig and then ran `p10k configure`.
In other JetBrains IDE, it works.  But now Pycharm is still.....

**Comment by falmar** [comment]
jetbrains terminal never played for my shell customization to the point i just deactivate it or open a terminal outside; you can check on the env variable the IDEs sets when you open the internal terminal... to make modifications or deactivate plugins

eg:
```bash
if [ "$TERMINAL_EMULATOR" != "JetBrains-JediTerm" ]; then
   // ...
fi
```

**Comment by flash-me** [comment]
Do you guys enabled by chance the "[New Terminal](https://blog.jetbrains.com/idea/2024/02/the-new-terminal-beta-is-now-in-jetbrains-ides/)"?

**Comment by NovemberZulu** [comment]
I don't have New UI enabled, so I don't even have an option to enable/disable New Terminal. After some experiments, it seems that New Terminal actually works better because it disables powerlever10k somehow? I didn't investigate deep, since I am too used to to "classic" IDEA UI.

**Comment by flash-me** [comment]
Do you have to correct font set in the IDE?

**Comment by jongwoo328** [comment]
It seems that this issue is related to this (https://youtrack.jetbrains.com/issue/IJPL-150357/powerline10k-theme-broken-in-terminal)

**Comment by euri10** [comment]
I had a "similar" issue I filed here: https://youtrack.jetbrains.com/issue/PY-73057/many-terminal-issues-output-1st-line-highlight-not-working

I managed to solve one of the issue (1st line was like prepended) but not the other (highlight not working) by running the configure again, the diff is here if anyone interested:
```
❯ diff -u .p10k.zsh .p10k.zsh.old
--- .p10k.zsh	2024-06-05 11:42:42.228947230 +0200
+++ .p10k.zsh.old	2023-12-07 19:04:02.565032499 +0100
@@ -1,7 +1,7 @@
-# Generated by Powerlevel10k configuration wizard on 2024-06-05 at 11:42 CEST.
-# Based on romkatv/powerlevel10k/config/p10k-lean.zsh, checksum 04242.
+# Generated by Powerlevel10k configuration wizard on 2023-12-07 at 19:04 CET.
+# Based on romkatv/powerlevel10k/config/p10k-lean.zsh, checksum 61526.
 # Wizard options: nerdfont-complete + powerline, small icons, unicode, lean, 24h time,
-# 1 line, compact, many icons, concise, transient_prompt, instant_prompt=verbose.
+# 1 line, sparse, many icons, fluent, transient_prompt, instant_prompt=verbose.
 # Type `p10k configure` to generate another config.
 #
 # Config for Powerlevel10k with lean prompt style. Type `p10k configure` to generate
@@ -83,7 +83,6 @@
     context                 # user@hostname
     nordvpn                 # nordvpn connection status, linux only (https://nordvpn.com/)
     ranger                  # ranger shell (https://github.com/ranger/ranger)
-    yazi                    # yazi shell (https://github.com/sxyazi/yazi)
     nnn                     # nnn shell (https://github.com/jarun/nnn)
     lf                      # lf shell (https://github.com/gokcehan/lf)
     xplr                    # xplr shell (https://github.com/sayanarijit/xplr)
@@ -137,7 +136,7 @@
   typeset -g POWERLEVEL9K_ICON_BEFORE_CONTENT=true
 
   # Add an empty line before each prompt.
-  typeset -g POWERLEVEL9K_PROMPT_ADD_NEWLINE=false
+  typeset -g POWERLEVEL9K_PROMPT_ADD_NEWLINE=true
 
   # Connect left prompt lines with these symbols.
   typeset -g POWERLEVEL9K_MULTILINE_FIRST_PROMPT_PREFIX=
@@ -489,7 +488,7 @@
   # Custom icon.
   # typeset -g POWERLEVEL9K_VCS_VISUAL_IDENTIFIER_EXPANSION='⭐'
   # Custom prefix.
-  # typeset -g POWERLEVEL9K_VCS_PREFIX='%fon '
+  typeset -g POWERLEVEL9K_VCS_PREFIX='%fon '
 
   # Show status of repositories of these types. You can add svn and/or hg if you are
   # using them. If you do, your prompt may become slow even when your current directory
@@ -550,7 +549,7 @@
   # Custom icon.
   # typeset -g POWERLEVEL9K_COMMAND_EXECUTION_TIME_VISUAL_IDENTIFIER_EXPANSION='⭐'
   # Custom prefix.
-  # typeset -g POWERLEVEL9K_COMMAND_EXECUTION_TIME_PREFIX='%ftook '
+  typeset -g POWERLEVEL9K_COMMAND_EXECUTION_TIME_PREFIX='%ftook '
 
   #######################[ background_jobs: presence of background jobs ]#######################
   # Don't show the number of background jobs.
@@ -720,12 +719,6 @@
   typeset -g POWERLEVEL9K_RANGER_FOREGROUND=178
   # Custom icon.
   # typeset -g POWERLEVEL9K_RANGER_VISUAL_IDENTIFIER_EXPANSION='⭐'
-  
-  ####################[ yazi: yazi shell (https://github.com/sxyazi/yazi) ]#####################
-  # Yazi shell color.
-  typeset -g POWERLEVEL9K_YAZI_FOREGROUND=178
-  # Custom icon.
-  # typeset -g POWERLEVEL9K_YAZI_VISUAL_IDENTIFIER_EXPANSION='⭐'
 
   ######################[ nnn: nnn shell (https://github.com/jarun/nnn) ]#######################
   # Nnn shell color.
@@ -915,7 +908,7 @@
   # Custom icon.
   # typeset -g POWERLEVEL9K_CONTEXT_VISUAL_IDENTIFIER_EXPANSION='⭐'
   # Custom prefix.
-  # typeset -g POWERLEVEL9K_CONTEXT_PREFIX='%fwith '
+  typeset -g POWERLEVEL9K_CONTEXT_PREFIX='%fwith '
 
   ###[ virtualenv: python virtual environment (https://docs.python.org/3/library/venv.html) ]###
   # Python virtual environment color.
@@ -1307,7 +1300,7 @@
   POWERLEVEL9K_KUBECONTEXT_DEFAULT_CONTENT_EXPANSION+='${${:-/$P9K_KUBECONTEXT_NAMESPACE}:#/default}'
 
   # Custom prefix.
-  # typeset -g POWERLEVEL9K_KUBECONTEXT_PREFIX='%fat '
+  typeset -g POWERLEVEL9K_KUBECONTEXT_PREFIX='%fat '
 
   ################[ terraform: terraform workspace (https://www.terraform.io) ]#################
   # Don't show terraform workspace if it's literally "default".
@@ -1351,7 +1344,7 @@
   #[ aws: aws profile (https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html) ]#
   # Show aws only when the command you are typing invokes one of these tools.
   # Tip: Remove the next line to always show aws.
-  typeset -g POWERLEVEL9K_AWS_SHOW_ON_COMMAND='aws|awless|cdk|terraform|pulumi|terragrunt'
+  typeset -g POWERLEVEL9K_AWS_SHOW_ON_COMMAND='aws|awless|terraform|pulumi|terragrunt'
 
   # POWERLEVEL9K_AWS_CLASSES is an array with even number of elements. The first element
   # in each pair defines a pattern against which the current AWS profile gets matched.
@@ -1536,7 +1529,7 @@
   # Custom icon.
   # typeset -g POWERLEVEL9K_TOOLBOX_VISUAL_IDENTIFIER_EXPANSION='⭐'
   # Custom prefix.
-  # typeset -g POWERLEVEL9K_TOOLBOX_PREFIX='%fin '
+  typeset -g POWERLEVEL9K_TOOLBOX_PREFIX='%fin '
 
   ###############################[ public_ip: public IP address ]###############################
   # Public IP color.
@@ -1639,7 +1632,7 @@
   # Custom icon.
   # typeset -g POWERLEVEL9K_TIME_VISUAL_IDENTIFIER_EXPANSION='⭐'
   # Custom prefix.
-  # typeset -g POWERLEVEL9K_TIME_PREFIX='%fat '
+  typeset -g POWERLEVEL9K_TIME_PREFIX='%fat '
 
   # Example of a user-defined prompt segment. Function prompt_example will be called on every
   # prompt if `example` prompt segment is added to POWERLEVEL9K_LEFT_PROMPT_ELEMENTS or
@@ -1652,7 +1645,7 @@
 
   # User-defined prompt segments may optionally provide an instant_prompt_* function. Its job
   # is to generate the prompt segment for display in instant prompt. See
-  # https://github.com/romkatv/powerlevel10k#instant-prompt.
+  # https://github.com/romkatv/powerlevel10k/blob/master/README.md#instant-prompt.
   #
   # Powerlevel10k will call instant_prompt_* at the same time as the regular prompt_* function
   # and will record all `p10k segment` calls it makes. When displaying instant prompt, Powerlevel10k
@@ -1688,7 +1681,7 @@
   #              it incompatible with your zsh configuration files.
   #   - quiet:   Enable instant prompt and don't print warnings when detecting console output
   #              during zsh initialization. Choose this if you've read and understood
-  #              https://github.com/romkatv/powerlevel10k#instant-prompt.
+  #              https://github.com/romkatv/powerlevel10k/blob/master/README.md#instant-prompt.
   #   - verbose: Enable instant prompt and print a warning when detecting console output during
   #              zsh initialization. Choose this if you've never tried instant prompt, haven't
   #              seen the warning, or if you are unsure what this all means.
  ~ ❯                                                                   
```

**Comment by NovemberZulu** [comment]
@flash-me yes, I have MesloLGS NF set both in iTerm2 and IDEA

@jongwoo328 That's it! If I launch IDEA from the dock, terminal works fine. However, I normally start IDEA from shell to make sure all environment variables are set properly.

All in all, I am not sure if it a Powerlevel10k bug or an IDEA bug :(

**Comment by NovemberZulu** [comment]
Created https://gitlab.com/gnachman/iterm2/-/issues/11594

**Comment by NovemberZulu** [comment]
As discovered in https://gitlab.com/gnachman/iterm2/-/issues/11594, there is a workaround: `TERMINFO_DIRS= idea` works fine.

**Comment by jaeho310** [comment]
This issue occurs when opening JetBrains IDE in iTerm.

Do not open in iterm
$ idea .
$ goland .
$ webstorm . 

Use default terminal or toolbox

**Comment by NovemberZulu** [comment]
The root cause was fixed in iTerm2 3.5.3
